algorithms/kernel/kmeans/kmeans_lloyd_helper.h

```diff
@@ -109,6 +110,47 @@ struct TlsTask
     size_t * cIndices         = nullptr;
 };
 
+template <Method method, typename algorithmFPType, CpuType cpu>
+struct BSHelper
+{
+    static size_t kmeansGetBlockSize(const size_t nRows, const size_t dim, const size_t clNum);
+};
+
+template <typename algorithmFPType, CpuType cpu>
+struct BSHelper<lloydDense, algorithmFPType, cpu>
+{
+    static size_t kmeansGetBlockSize(const size_t nRows, const size_t dim, const size_t clNum)
```

Comment:
Now it's okay. But we need to think about extending this for any DAAL algorithm.
I suppose it will be applicable for large list of algorithms if we will set some function which compute memory consumption based on nRows.